nbgitpuller/handlers.py

```diff
@@ -152,12 +154,14 @@ def get(self):
         subPath = self.get_argument('subpath', None) or \
                   self.get_argument('subPath', '.')
         app = self.get_argument('app', app_env)
+        parent_reldir = os.getenv('NBGITPULLER_PARENTPATH', '')
+        targetpath = self.get_argument('targetpath', None) or \
+                     self.get_argument('targetPath', repo.split('/')[-1])
 
         if urlPath:
             path = urlPath
         else:
-            repo_dir = repo.split('/')[-1]
-            path = os.path.join(repo_dir, subPath)
+            path = os.path.join(parent_reldir, targetpath, subPath)
```

Comment:
Do we want to enforce any restrictions on this, like ensuring targetPath can't be a parent of the CWD or something? Or is this a "use at your own risk" kinda thing?

Reply:
Most people will use the default destination and won't set `targetPath`. Since this is for advanced users I think it's fine to let them set this to any path including those outside the working directory if they know what they're doing.